src/simplejavacalculator/UI.java

```diff
@@ -217,94 +215,88 @@ public void init() {
    }
    
    @Override
    public void actionPerformed(ActionEvent e) {
       final Object source = e.getSource();
-      
+      Double checkNum = null;
+
       for (int i = 0; i < 10; i++) {
          if (source == but[i]) {
             text.replaceSelection(buttonValue[i]);
             return;
          }
       }
-      
-      if (source == butAdd) {
-         writer(calc.calculateBi(Calculator.BiOperatorModes.add, reader()));
-      }
-      
-      if (source == butMinus) {
-         writer(calc.calculateBi(Calculator.BiOperatorModes.minus, reader()));
-      }
-      
-      if (source == butMultiply) {
-         writer(calc.calculateBi(Calculator.BiOperatorModes.multiply,
-                                 reader()));
-      }
-      
-      if (source == butDivide) {
-         writer(calc
-                   .calculateBi(Calculator.BiOperatorModes.divide, reader()));
-      }
-      if (source == butxpowerofy) {
-         writer(calc
-                   .calculateBi(Calculator.BiOperatorModes.xpowerofy, reader()));
-      }
-      
-      if (source == butSquare) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.square,
-                                   reader()));
-      }
-      
-      if (source == butSquareRoot) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.squareRoot,
-                                   reader()));
-      }
-      
-      if (source == butOneDividedBy) {
-         writer(calc.calculateMono(
-                                   Calculator.MonoOperatorModes.oneDividedBy, reader()));
-      }
-      
-      if (source == butCos) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.cos,
-                                   reader()));
-      }
-      
-      if (source == butSin) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.sin,
-                                   reader()));
-      }
-      
-      if (source == butTan) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.tan,
-                                   reader()));
-      }
-      if (source == butlog) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.log,
-                                   reader()));
-      }
-      if (source == butrate) {
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.rate,
-                                   reader()));
-      }
-      if(source == butabs){
-         writer(calc.calculateMono(Calculator.MonoOperatorModes.abs, reader()));
-      }
-      
-      if (source == butEqual) {
-         writer(calc.calculateEqual(reader()));
-      }
-      
-      if (source == butCancel) {
-         writer(calc.reset());
+
+      try {
+         checkNum = Double.parseDouble(text.getText());
+      } catch(NumberFormatException k){
+
       }
-      
-      if (source == butBinary) {
-         parsetoBinary();
+
+      if(checkNum != null || source == butCancel){
+         if (source == butAdd) {
```

Comment:
Hm, the following would probably need to be moved into a `switch` statement (and idealy be refactored into its own method..).